### PR TITLE
NS2.0 - Complete System.IO.IsolatedStorage namespace

### PR DESCRIPTION
**Pull request by AtsushiKan**
IsolatedStorageFile:

  Quota/MaximumSize
    long.MaxValue (as desktop does when there's no quota)

  UsedSize/CurrentSize:
    PNSE (there's no mechanism in this implementation to
      track usage, and I'm assuming we don't plan to
      add one...)

    I'd also be open to returning 0 from both of these -
    they're only useful for checking if you're running
    up against a quota.

  AvailableFreeSpace
    Quota - UsedSpace  (just to stay in sync with desktop)

IsolatedStorageFileStream:

  SafeFileHandle
    throw ISFException (as in desktop - this is now allows)

  Flush(bool)
  IsAsync
    Obvious delegation.

### CODE REVIEW

**Comment by AtsushiKan** [comment]
cc @JeremyKuhne

File: src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs [hunk 1]
```diff
@@ -504,6 +504,50 @@ public void Reset()
...
+            {
+                return long.MaxValue;
+            }
+        }
+
+        public override long UsedSize
+        {
+            get
+            {
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_IsolatedStorageSize);
```
Comment (JeremyKuhne) [inline src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs]: I think if we're not going to have Quota throw we should probably do 0 for Current/UsedSize.

Comment (AtsushiKan) [inline src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs]: Cool by me.

**Comment by joshfree** [comment]
Thanks @AtsushiKan 😄

**Pull request merged**